drivers/adc/adc_mcux_adc12.c

```diff
@@ -252,26 +253,26 @@ static const struct adc_driver_api mcux_adc12_driver_api = {
 #define TO_ADC12_CLOCK_SRC(val) _DO_CONCAT(kADC12_ClockSourceAlt, val)
 #define TO_ADC12_CLOCK_DIV(val) _DO_CONCAT(kADC12_ClockDivider, val)
 
-#if DT_INST_0_NXP_KINETIS_ADC12
+#if DT_HAS_NODE(DT_INST(0, nxp_kinetis_adc12))
```

Comment:
Can we use `DT_DRV_COMPAT` definition here?

Reply:
Introduced `DT_HAS_DRV_INST`, now its `#if DT_HAS_DRV_INST(0)`